components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/IdentityProvider.java

```diff
@@ -365,19 +384,179 @@ public void setAlias(String alias) {
     }
 
     /**
-     * @return
+     * Get certificate
+     * @return if certificate is in JSON array then return only first element.
      */
     public String getCertificate() {
+
+        // Check whether certificate is in json format.
+        if (StringUtils.isNotBlank(certificate) && certificate.startsWith(JSON_ARRAY_IDENTIFIER)) {
+            if (!certificate.equals(EMPTY_JSON_ARRAY)) {
+                certificate = ((JSONObject) (new JSONArray(certificate).get(0))).getString(CERT_VALUE);
+            }
+            // If certificate is an empty json array, then return empty value.
+            else {
+                certificate = "";
+            }
+        }
         return certificate;
     }
 
     /**
      * @param certificate
      */
     public void setCertificate(String certificate) {
+
+        setCertificateInfoArray(certificate);
         this.certificate = certificate;
     }
 
+    /**
+     * Get certificate info array
+     * @return CertificateInfo array
+     */
+    public CertificateInfo[] getCertificateInfoArray() {
+
+        return certificateInfoArray;
+    }
+
+    /**
+     * Set certificateInfoArray
+     *
+     * @param certificateValue array which contains certificate info. Here Certificate info contains thumbPrint
+     *                             and certificate value as attributes.
+     */
+    private void setCertificateInfoArray(String certificateValue) {
+
+        try {
+            if (StringUtils.isNotBlank(certificateValue) && !certificateValue.equals(EMPTY_JSON_ARRAY)) {
+                certificateValue = certificateValue.trim();
+                try {
+                    this.certificateInfoArray = handleJsonFormatCertificate(certificateValue);
+                } catch (JSONException e) {
+                    if ("A JSONArray text must start with '[' at 1 [character 2 line 1]".equals(e.getMessage())) {
```

Comment:
can take, plain text and encoded certificate handling outside the "if ("A JSONArray te.."